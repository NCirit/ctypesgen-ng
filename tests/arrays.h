typedef double my_complex[2];

int my_main(int argc, char *argv[]);

typedef double my_complex_3d_point[3][2];

typedef double _Complex complex_3d_point[3];

typedef __int128 big_t[2];
