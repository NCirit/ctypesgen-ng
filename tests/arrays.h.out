import ctypes
import ctypes.util
import os

os.environ['LIBRARY_PATH'] = os.path.dirname(os.path.abspath(__file__))
test_path = ctypes.util.find_library('test')
test = ctypes.CDLL(os.path.abspath(test_path))

typedef_my_complex = (ctypes.c_double * 2)

test.my_main.restype = ctypes.c_int
test.my_main.argtypes = [ctypes.c_int, ctypes.POINTER(ctypes.c_char_p)]

typedef_my_complex_3d_point = ((ctypes.c_double * 2) * 3)

typedef_complex_3d_point = ((ctypes.c_double * 2) * 3)

typedef_big_t = ((ctypes.c_ubyte * 16) * 2)
